### PR TITLE
Post quote embeds for Discord message links

### LINKED ISSUE #145: Reply with an embed of the message when a message link is seen
an extension to !paste, when any message url is posted in chat, Modix should try to parse it and send an embed containing the quoted message:
https://discordapp.com/channels/143867839282020352/364492811489509376/480464494863187969

### PR DESCRIPTION
**Pull request by jrusbatch**
Fixes #145.

### CODE REVIEW

File: Modix.Services/Quote/MessageLinkBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+using Discord.WebSocket;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Modix.Services.Quote
+{
+    public class MessageLinkBehavior : BehaviorBase
+    {
+        private static readonly Regex Pattern = new Regex(
+            @"https://(canary\.)?discordapp\.com/channels/(?<GuildId>\d+)/(?<ChannelId>\d+)/(?<MessageId>\d+)",
```
Comment (Scott-Caldwell) [inline Modix.Services/Quote/MessageLinkBehavior.cs]: It appears as though regular `http` also works, so maybe make the `s` optional

File: Modix.Services/Quote/MessageLinkBehavior.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        protected internal override Task OnStoppedAsync()
+        {
+            DiscordClient.MessageReceived -= OnMessageReceived;
+
+            return Task.CompletedTask;
+        }
+
+        private async Task OnMessageReceived(SocketMessage message)
```
Comment (Scott-Caldwell) [inline Modix.Services/Quote/MessageLinkBehavior.cs]: Nitpick: should this be suffixed with "Async"?

File: Modix.Services/Quote/MessageLinkBehavior.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            if (!(message is SocketUserMessage userMessage) ||
+                !(userMessage.Author is SocketGuildUser || userMessage.Author.IsBot) ||
```
Comment (Scott-Caldwell) [inline Modix.Services/Quote/MessageLinkBehavior.cs]: Having trouble parsing this line. Is this line supposed to be `!(userMessage.Author is SocketGuildUser) || userMessage.Author.IsBot ||`?

Comment (Scott-Caldwell) [inline Modix.Services/Quote/MessageLinkBehavior.cs]: Should we exclude bots? Just thinking about a use case where someone might put a message link in a tag or something and expect a quoted reply.

If we do decide to exclude bots, it probably makes sense to exclude webhooks as well.

File: Modix.Services/Quote/MessageLinkBehavior.cs (hunk 1 above)
Comment (jmazouri) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Official links from the client will always be `https`, but yeah.

File: Modix.Services/Quote/MessageLinkBehavior.cs (hunk 3 above)
Comment (jmazouri) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: We should probably exclude bots as some combination of commands may result in an infinite loop. Webhooks too, unless needed.

Comment (Scott-Caldwell) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Good point, didn't think about that

Comment (jrusbatch) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: This evaluates to `false` if the message author is not a `SocketGuildUser` or if the author is a `SocketGuildUser` but is also bot. I can refactor it to try to make it more clear.

Comment (jrusbatch) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Good catch - updated.

Comment (jrusbatch) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Is [this](https://github.com/discord-csharp/MODiX/pull/221/commits/3ec9a9d62b69230d5d7ebbd3bf8333f996ac378c) better?

File: Modix.Services/Quote/MessageLinkBehavior.cs (hunk 2 above)
Comment (jrusbatch) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Yep, updated.

File: Modix.Services/Quote/MessageLinkBehavior.cs (hunk 3 above)
Comment (Scott-Caldwell) [reply Modix.Services/Quote/MessageLinkBehavior.cs]: Much clearer. So simple that even I can understand it :)

**Pull request merged**